pkgs/cli_util/lib/src/base_directories.dart

```diff
@@ -0,0 +1,274 @@
+  /// typically containing only lower-case characters and underscores to ensure
+  /// maximum compatibility and readability. For example, "my_dart_tool".
+  final String tool;
+
+  /// The environment variables to use.
+  ///
+  /// Defaults to [Platform.environment].
+  final Map<String, String> environment;
+
+  /// Constructs a [BaseDirectories] instance for the given [tool] name.
+  BaseDirectories({
+    required this.tool,
+    Map<String, String>? environment,
+  }) : environment = environment ?? Platform.environment;
+
+  /// The path of the directory where [tool] can place its caches.
+  ///
+  /// The cache might at any point be purged by the operating system or user.
+  /// Applications should be able to reconstruct any data stored here. If [tool]
+  /// cannot handle data being purged, use [runtimeHome] or [dataHome] instead.
+  ///
+  /// This is a location appropriate for storing non-essential files that may be
+  /// removed at any point.
+  ///
+  /// The directory location depends on the current [Platform.operatingSystem]:
+  /// - on **Windows**:
+  ///   - `%LOCALAPPDATA%\<tool>`
+  /// - on **Mac OS**:
+  ///   - `$HOME/Library/Caches/<tool>`
+  /// - on **Linux**:
+  ///   - `$XDG_CACHE_HOME/<tool>` if `$XDG_CACHE_HOME` is defined, and
+  ///   - `$HOME/.cache/<tool>` otherwise.
+  ///
+  /// The directory won't be created, the method merely returns the recommended
+  /// location.
+  ///
+  /// On some platforms, this path may overlap with [runtimeHome] and
+  /// [stateHome].
+  ///
+  /// Throws an [EnvironmentNotFoundException] if the necessary environment
+  /// variables are undefined.
+  String cacheHome() =>
+      path.join(_baseDirectory(_XdgBaseDirectoryType.cache), tool);
+
+  /// The path of the directory where [tool] can place its configuration.
+  ///
+  /// The configuration may be synchronized across devices by the OS and may
+  /// survive application removal.
+  ///
+  /// This is a location appropriate for storing application specific
+  /// configuration for the current user.
+  ///
+  /// The directory location depends on the current [Platform.operatingSystem]
+  /// and what file types are stored:
+  /// - on **Windows**:
+  ///   - `%APPDATA%\<tool>`
+  /// - on **Mac OS**:
+  ///   - `$HOME/Library/Preferences/<tool>` (may only contain `.plist` files)
+  ///     if [plistFiles] is true, and
+  ///   - `$HOME/Library/Application Support/<tool>` otherwise.
+  /// - on **Linux**:
+  ///   - `$XDG_CONFIG_HOME/<tool>` if `$XDG_CONFIG_HOME` is defined, and
+  ///   - `$HOME/.config/<tool>` otherwise.
+  ///
+  /// The directory won't be created, the method merely returns the recommended
+  /// location.
+  ///
+  /// On some platforms, this path may overlap with [dataHome].
+  ///
+  /// Throws an [EnvironmentNotFoundException] if the necessary environment
+  /// variables are undefined.
+  String configHome({bool plistFiles = false}) {
```

Comment:
It's a little annoying that this is a function and the others are getters.
Not enough to make the others not be getters.

Could this flag be a separate getter, a `String get plistConfigHome` getter on the side, which works the same on non-MacOS, and there it gives the plist-only directory?